### PR TITLE
Fix out ouf bounds when taking from run end arrays

### PR DESCRIPTION
**Pull request by robert3005**

### CODE REVIEW

**Comment by robert3005** [comment]
Still need to write a test

**Comment by a10y** [comment]
thank you

**Review by a10y — APPROVED** [review]
cherrypicked this to my branch and the benchmarks work again

**Comment by robert3005** [comment]
But something is wrong since the tests fail which makes me question my sanity

**Comment by robert3005** [comment]
Ah this has to work the same way sparse array works

File: vortex-array/src/compute/search_sorted.rs [hunk 1]
```diff
@@ -27,7 +27,20 @@ impl SearchResult {
         }
     }
 
-    pub fn to_index(self) -> usize {
+    pub fn to_non_zero_offset_index(self, len: usize) -> usize {
```
Comment (robert3005) [inline vortex-array/src/compute/search_sorted.rs:30]: need to think about better names for this, `to_index_end_exclusive`?

**Pull request merged**